### PR TITLE
Follow-up changes for 002-login-error-message branch

### PR DESCRIPTION
**Pull request by mhilton**
This adds tests and some changes suggested in #30

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore-client/53/
Test PASSed.

File: cmd/charm/charmcmd/cmd.go [hunk 1]
```diff
@@ -203,30 +203,32 @@ func ussoTokenPath() string {
...
+// returned unchanged.
+func translateError(err error) error {
 	if err == nil {
 		return err
 	}
 	cause := errgo.Cause(err)
 	switch {
 	case httpbakery.IsInteractionError(cause):
-		return interactionErrorMessage(ctxt, cause.(*httpbakery.InteractionError))
+		return translateInteractionError(cause.(*httpbakery.InteractionError))
```
Comment (rogpeppe) [inline cmd/charm/charmcmd/cmd.go]: How about:

```
err := translateInteractionError(cause.(*http.InteractionError))
return errgo.Notef(err, "login failed")
```

then we can get rid of a lot of indentation in the function below by returning early.

File: cmd/charm/charmcmd/cmd.go [hunk 2]
```diff
@@ -203,30 +203,32 @@ func ussoTokenPath() string {
...
-// and outputs the message ctxt.Stderr.
-func interactionErrorMessage(ctxt *cmd.Context, err *httpbakery.InteractionError) error {
-	reason := err.Reason.Error()
-	if ussoError, ok := errgo.Cause(err.Reason).(*usso.Error); ok {
-		reason = ussoError.Error()
+// translateInteractionError translates err into a new error with a user
+// understandable error message.
+func translateInteractionError(err *httpbakery.InteractionError) error {
+	reason := err.Reason
+	if ussoError, ok := errgo.Cause(reason).(*usso.Error); ok {
```
Comment (rogpeppe) [inline cmd/charm/charmcmd/cmd.go]: ussoError, ok := errgo.Cause(reason).(*usso.Error)
if !ok {
      return err
}

File: cmd/charm/charmcmd/cmd.go [hunk 3]
```diff
@@ -203,30 +203,32 @@ func ussoTokenPath() string {
...
+		reason = ussoError
 		if ussoError.Code == "INVALID_DATA" && len(ussoError.Extra) > 0 {
```
Comment (rogpeppe) [inline cmd/charm/charmcmd/cmd.go]: if ussoError.Code != "INVALID DATA" || len(ussoError.Extra) == 0 {
     return err
}

File: cmd/charm/charmcmd/cmd.go [hunk 4]
```diff
@@ -234,10 +236,9 @@ func interactionErrorMessage(ctxt *cmd.Context, err *httpbakery.InteractionError
 				if v1, ok := v.([]interface{}); ok && len(v1) > 0 {
 					v = v1[0]
 				}
-				reason = fmt.Sprintf("%s: %s", k, v)
+				reason = errgo.Newf("%s: %s", k, v)
```
Comment (rogpeppe) [inline cmd/charm/charmcmd/cmd.go]: return here?

**Comment by rogpeppe** [comment]
LGTM with a suggestion for losing a bunch of indentation. Thanks.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore-client/54/
Test PASSed.

**Comment by frankban** [comment]
:+1: nice thank you!

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore-client/55/
Test PASSed.

**Comment by mhilton** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-client-merge

**Pull request merged**